### PR TITLE
CLOUDSTACK-8785: Proper enforcement of retry count (max.retries) for …

### PR DESCRIPTION
**Pull request by koushik-das**
…all work type handled by HighAvailability manager

Retry count is properly enforced for all work types in HA manager. Also reorganized some of the code for easy testing.

### CODE REVIEW

**Comment by koushik-das** [comment]
Anyone? @kishankavala @bhaisaab @remibergsma @DaanHoogland

File: server/src/com/cloud/ha/HighAvailabilityManagerImpl.java [hunk 1]
```diff
@@ -383,10 +383,10 @@ public void scheduleRestart(VMInstanceVO vm, boolean investigate) {
         }
 
         List<HaWorkVO> items = _haDao.findPreviousHA(vm.getId());
-        int maxRetries = 0;
+        int timesTried = 0;
```
Comment (DaanHoogland) [inline server/src/com/cloud/ha/HighAvailabilityManagerImpl.java]: I like the rename!

**Comment by DaanHoogland** [comment]
in terms of refactorring i like this very much. unit tests and travis pass so LGTM. I haven't done any integration testing, have you @koushik-das

**Comment by bhaisaab** [comment]
followed the refactored logic, LGTM

**Comment by koushik-das** [comment]
@DaanHoogland I ran the following integration test on simulator. Other than this I don't thick there is any other test for VM HA.

test/integration/smoke/misc/test_vm_ha.py

Test VM HA ... === TestName: test_vm_ha | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 258.037s

OK

**Comment by koushik-das** [comment]
2 LGTMs merging.

**Pull request merged**